Go generation of protos during install is problematic

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Debian Buster
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.5.0-rc2
- Python version: 3.8
- Installed using virtualenv? pip? conda?: n/a
- Bazel version (if compiling from source): 3.7.2
- GCC/Compiler version (if compiling from source): 8.3.0-6
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

**Describe the problem**

There exists a general issue, of which this specific issue contributes, that installation and use of Tensorflow in Golang is complicated. Use of `go get` to obtain Tensorflow fails.  The reason for this is that the protos need to be generated; see #44655 and discussion that including generated protos would create a maintenance issue.  To resolve this, a user must fetch the repo and then issue the following command:

```sh
go generate github.com/tensorflow/tensorflow/tensorflow/go/op
```

This results in two problems:

1. The `go generate` command errors and completes with a non-zero exit status:

```
../genop/internal/api_def_map.go:34:2: no required module provides package github.com/tensorflow/tensorflow/tensorflow/go/core/framework/api_def_go_proto; to add it:
        go get github.com/tensorflow/tensorflow/tensorflow/go/core/framework/api_def_go_proto
../genop/internal/api_def_map.go:35:2: no required module provides package github.com/tensorflow/tensorflow/tensorflow/go/core/framework/op_def_go_proto; to add it:
        go get github.com/tensorflow/tensorflow/tensorflow/go/core/framework/op_def_go_proto
generate.go:18: running "go": exit status 1
```

2. The generated protos are written to a vendor directory (`tensorflow/go/vendor`), although Go will not search for them there. The `vendor` directory is not meant for adding arbitrary modules (see [https://github.com/golang/go/issues/29079](https://github.com/golang/go/issues/29079)).  Thus, the contents must be manually relocated from the vendor directory to `tensorflow/go`.

### Proposed resolution

To make installation more straightforward, I propose that the protos be output in their correct location with the source tree (not in `vendor`).  Indeed, `tensorflow/go/op/generate.go` does this to some extent now, overwriting `wrappers.go` in the source tree.

--- a/tensorflow/go/genop/generate.sh
+++ b/tensorflow/go/genop/generate.sh
@@ -61,12 +61,11 @@ fi
 # Ensure that protoc-gen-go is available in $PATH
 # Since ${PROTOC} will require it.
 export PATH=$PATH:${GOPATH}/bin
-mkdir -p ../vendor
 for FILE in ${TF_DIR}/tensorflow/core/framework/*.proto \
     ${TF_DIR}/tensorflow/core/protobuf/*.proto \
     ${TF_DIR}/tensorflow/stream_executor/*.proto; do
   ${PROTOC} \
     -I ${TF_DIR} \
-    --go_out=../vendor \
+    --go_out=${GOPATH}/src \
     $FILE
 done

